### ISSUE TITLE
【Feature】 Allows creation of events from callback.

### ISSUE THREAD

**Issue by not-elm** [elm-register/bevy-input-sequence#5]
I would like to be able to generate events from callbacks with the proposal mentioned in the PR#4.
This allows gamepad buttons to be passed to events.

```rust
   commands.spawn(ButtonSequence::new(
        [
            GamepadButtonType::North,
            GamepadButtonType::East,
            GamepadButtonType::South,
            GamepadButtonType::West,
        ],
        |gamepad: Gamepad|{
            MyEvent(gamepad)
        }
    ).time_limit(Duration::from_secs(1)));
```

**Comment by not-elm** [comment]
PS Controller have been broken, so it may take some time to test .

**Comment by shanecelis** [comment]
I got you, fam. See PR https://github.com/not-elm/bevy-input-sequence/pull/6.

```rust
 commands.add(ButtonSequence::new(
         action::send_event_with_input(|gamepad| MyEvent(gamepad)),
        [
            GamepadButtonType::North,
            GamepadButtonType::East,
            GamepadButtonType::South,
            GamepadButtonType::West,
        ]
    ).time_limit(Duration::from_secs(1)));
```

I thought about switching the arguments to `ButtonSequence::new()` and `KeySequence::new()` too. Looks good for `ButtonSequence`. Looks a little redundant for `KeySequence` because it reads: 

```rust
KeySequence::new(keyseq! { ctrl-alt-Delete }, quit_everything)
```

But maybe that's ok.

**Comment by not-elm** [comment]
I would like to update this library. 0.3.1 is fine?

**Comment by shanecelis** [comment]
I’d suggest 0.4.0 because it’s a breaking change.

**Comment by not-elm** [comment]
I see, thank you.

**Comment by not-elm** [comment]
I released version 0.4.0!
This issue close.

[Release note](https://github.com/not-elm/bevy-input-sequence/releases/tag/release-v0.4.0)
